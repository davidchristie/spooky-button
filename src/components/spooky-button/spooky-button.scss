@import url('https://fonts.googleapis.com/css?family=Butcherman');

button {
  border: none;
  border-radius: 5px;
  box-shadow: 0 0 5px black;
  cursor: not-allowed;
  font-size: 24px;
  padding: 10px;
  user-select: none;
  text-shadow: 0 0 1px black;
  transition: background-color 500ms ease;
  transition: box-shadow 500ms ease;
  transition: text-shadow 500ms ease;
  vertical-align: middle;

  &.clicked {
    background-color: black;
    box-shadow: 0 0 4px 5px black;
    color: white;
    font-family: 'Butcherman', cursive;
    height: 100vh;
    text-shadow: 0 0 10px white;
    width: 100vw;

    .skull {
      opacity: 1;
      visibility: visible;
    }
  }

  &:hover {
    background-color: black;
    box-shadow: 0 0 4px 5px black;
    color: white;
    font-family: 'Butcherman', cursive;
    text-shadow: 0 0 10px white;

    .skull {
      opacity: 1;
      visibility: visible;
    }
  }

  .skeleton {
    display: block;
    position: fixed;
  }

  .skull {
    font-size: 30px;
    opacity: 0;
    padding: 10px;
    transition: opacity 500ms ease;
  }
}
