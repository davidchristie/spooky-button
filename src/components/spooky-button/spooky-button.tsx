import { Component, State } from '@stencil/core'

@Component({
  tag: 'spooky-button',
  styleUrl: 'spooky-button.scss'
})
export class SpookyButton {

  @State() clicked : boolean = false

  private onClick = () => {
    this.clicked = true
  }

  render () {
    return (
      <button class={this.clicked ? 'clicked' : null} onClick={this.onClick}>
        <span class='skull'>ğŸ’€</span>
        <slot />
        <span class='skull'>ğŸ’€</span>
        {
          this.clicked
            ? <spooky-skeletons></spooky-skeletons>
            : null
        }
      </button>
    );
  }
}
