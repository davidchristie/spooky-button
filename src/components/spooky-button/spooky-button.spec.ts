import { flush, render } from '@stencil/core/testing'
import { SpookyButton } from './spooky-button'

describe('spooky-button', () => {
  it('should build', () => {
    expect(new SpookyButton()).toBeTruthy()
  })

  describe('rendering', () => {
    it('without content', async () => {
      const element = await render({
        components: [SpookyButton],
        html: '<spooky-button></spooky-button>'
      })
      expect(element).toMatchSnapshot()
    })

    it('with content', async () => {
      const element = await render({
        components: [SpookyButton],
        html: `
          <spooky-button>
            Test content
          </spooky-button>
        `
      })
      expect(element).toMatchSnapshot()
    })

    it('on click', async () => {
      const element = await render({
        components: [SpookyButton],
        html: '<spooky-button></spooky-button>'
      })
      element.querySelector('button').click()
      await flush(element)
      expect(element).toMatchSnapshot()
    })
  })
})
