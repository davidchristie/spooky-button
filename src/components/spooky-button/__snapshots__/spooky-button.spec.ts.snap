// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spooky-button rendering on click 1`] = `
<spooky-button
  class="hydrated"
>
  <button
    class="clicked"
  >
    <span
      class="skull"
    >
      ðŸ’€
    </span>
    <!---->
    <span
      class="skull"
    >
      ðŸ’€
    </span>
    <spooky-skeletons />
  </button>
</spooky-button>
`;

exports[`spooky-button rendering with content 1`] = `
<spooky-button
  class="hydrated"
>
  <button
    class=""
  >
    <span
      class="skull"
    >
      ðŸ’€
    </span>
    <!---->
    
            Test content
          
    <span
      class="skull"
    >
      ðŸ’€
    </span>
    
  </button>
</spooky-button>
`;

exports[`spooky-button rendering without content 1`] = `
<spooky-button
  class="hydrated"
>
  <button
    class=""
  >
    <span
      class="skull"
    >
      ðŸ’€
    </span>
    <!---->
    <span
      class="skull"
    >
      ðŸ’€
    </span>
    
  </button>
</spooky-button>
`;
